### PR TITLE
build: Migrate views to Phoenix 1.7

### LINKED ISSUE #339: Migrate to Phoenix 1.7

### PR DESCRIPTION
**Pull request by ruioliveira02**
Closes #339.

Admin controllers aren't migrated. I think we should remove them and start from scratch there. What do you think @RuiL1904 @MarioRodrigues10 ?

### CODE REVIEW

**Comment by RuiL1904** [comment]
@ruioliveira02 I agree.

File: lib/safira_web/controllers/fallback_controller.ex [hunk 1]
```diff
@@ -32,14 +34,19 @@ defmodule SafiraWeb.FallbackController do
...
+    |> put_view(json: ErrorJSON)
     |> put_status(:unprocessable_entity)
-    |> render(SafiraWeb.ChangesetView, "error.json", changeset: changeset)
+    |> render(:error, changeset: changeset)
   end
 
   def call(conn, {:error, :not_found}) do
+    require Logger
+    Logger.warn("BATATA")
+
```
Comment (MarioRodrigues10) [inline lib/safira_web/controllers/fallback_controller.ex:45]:
```suggestion


```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

File: lib/safira_web/controllers/attendee/attendee_json.ex [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      nickname: at.nickname,
+      name: at.name,
+      avatar: Avatar.url({at.avatar, at}, :original),
+      cv: CV.url({at.cv, at}, :original),
+      badges: for(b <- at.badges, do: BadgeJSON.badge(%{badge: b})),
+      badge_count: at.badge_count,
+      token_balance: at.token_balance,
+      prizes: for(p <- at.prizes, do: PrizeJSON.prize_attendee(%{prize: p})),
+      entries: at.entries,
+      redeemables: for(r <- at.redeemables, do: RedeemableJSON.my_readeemables(%{redeemable: r}))
```
Comment (MarioRodrigues10) [inline lib/safira_web/controllers/attendee/attendee_json.ex:40]:
```suggestion
      redeemables: for(r <- at.redeemables, do: RedeemableJSON.my_redeemables(%{redeemable: r}))
```
Fix the typo in this line

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]
It returns the following error:
image
image

File: lib/safira_web/controllers/roulette/roulette_json.ex [hunk 3]
```diff
@@ -30,18 +33,14 @@ defmodule SafiraWeb.RouletteView do
     resp
   end
 
-  def render("price.json", %{price: price}) do
-    %{price: price}
-  end
```
Comment (MarioRodrigues10) [inline lib/safira_web/controllers/roulette/roulette_json.ex:35]: I think is due to this function you should create
```elixir
def price(%{price: price}) do
 %{price: price}
end

```

**Comment by joaodiaslobo** [comment]
No longer applicable.

**Pull request closed without merging**